src/MouseAdapter.js

```diff
@@ -309,20 +352,231 @@ define(function(require) {
+
+            // Reset the delta arrays on MouseEnd to prevent false positives between
+            // natural end and start events.
+            this._deltaArrayX = [];
+            this._deltaArrayY = [];
+
+            this.onMouseWheelEnd.dispatch([{
+                distance: { x: 0, y: 0 },
+                source: event
+            }]);
+            this._wheeling = false;
+        },
+
+
+        /**
+         * Perform a ReScroll dispatch, which currently means dispatch a MouseEnd event 
+         * followed by a MouseStart event, to alert listeners that the user performed a
+         * new distinct scroll event.  This could easily be refactored into a separate
+         * observable if desired in the future.
+         * @param {object} event
+         * @private
+         */
+        _dispatchUserReScroll: function(event) {
             this._dispatchMouseWheelEnd(event);
+            this._dispatchMouseWheelStart(event);
+        },
+
+        /**
+         * This is a helper function which takes in a set of consecutive linear data 
+         * and tries to determine whether it is increasing or decreasing.  It can
+         * operate on an entire array, or only a specified section of it. When given a
+         * starting point and a length that extends outside the boundaries of the array,
+         * the function assumes the data wraps around to the beginning of the array.
+         * @param {Array.<number>} array The array of data to perform analysis on.
+         * @param {number} arrayMaxLength [optional] The max length of the array.  If
+         * not given, assumed to be array.length
+         * @param {number} startAt [optional] The index in the array where the target 
+         * data begins.  If not specified, assumes this index is 0.
+         * @param {number} length [optional] The number of data points to examine.  
+         * If not specified, assumed to be the entire array length.
+         * @return {number} Returns:
+                1 if data appears to be increasing.
+               -1 if data appears to be decreasing.
+                0 if there is no discernable trend.
+         * @private
+         */
+        _calculateDeltaDirection: function(array, startAt, length, arrayMaxLength) {
+            arrayMaxLength = arrayMaxLength || array.length;
+            startAt = startAt || 0;
+            length = length || arrayMaxLength;
+            var total = 0;
+            var offsetIdx1,offsetIdx2;
+            for (var idx1 = 0; idx1 < length; idx1++) {
+                for (var idx2 = idx1+1; idx2 < length; idx2++) {
+                    offsetIdx1 = (idx1+startAt) % arrayMaxLength;
+                    offsetIdx2 = (idx2+startAt) % arrayMaxLength;
+                    if (array[offsetIdx1] < array[offsetIdx2]) {
+                        total += (1/(idx2-idx1));
+                    }
+                    if (array[offsetIdx1] > array[offsetIdx2]) {
+                        total += (-1/(idx2-idx1));
```

Comment:
So, basically, the effect on direction lessens the farther away the delta is, at a rate of 1/distance? That seems like a useful comment to include (took me a while to think it though).